packages/ubus-lime-location/files/usr/libexec/rpcd/lime-location

```diff
@@ -32,43 +25,40 @@ end
 
 local function _get_location(msg)
     local result = {location = {}, default = false}
-    local lat = conn:call("uci", "get", {config="libremap", section="location", option="latitude" }).value
-    local lon = conn:call("uci", "get", {config="libremap", section="location", option="longitude" }).value
+    local uci = config.get_uci_cursor()
+
+    local lat = uci:get("libremap", "location", "latitude")
+    local lon = uci:get("libremap", "location", "longitude")
 
     if (type(tonumber(lat)) == "number" and type(tonumber(lon)) == "number") then
         result.location.lat = lat
         result.location.lon = lon
     else
-        result.location.lat = conn:call("uci", "get", {config="libremap", section="@libremap[0]",
-                                              option="community_lat" }).value
-        result.location.lon = conn:call("uci", "get", {config="libremap", section="@libremap[0]",
-                                              option="community_lon" }).value
-        result.default = true                                              
+        result.location.lat = uci:get("libremap", "settings", "community_lat")
+        result.location.lon = uci:get("libremap", "settings", "community_lon")
+        result.default = true
     end
+
+    result.status = "ok"
     return result;
 end
 
 local function get_location(msg)
-  printJson(_get_location(msg))
+  utils.printJson(_get_location(msg))
 end
 
 local function set_location(msg)
-    local result = {}
-    conn:call("uci", "set", {config="libremap", section="location", values={ longitude = msg.lon, latitude = msg.lat } })
-    printJson({ lat = msg.lat, lon = msg.lon });
-end
+    local uci = config.get_uci_cursor()
+    uci:set("libremap", "location", "latitude", msg.lon)
+    uci:set("libremap", "location", "longitude", msg.lat)
 
-local function lines(str)
-  -- split a string into lines separated by line endings
-  local t = {}
-  local function helper(line) table.insert(t, line) return "" end
-  helper((str:gsub("(.-)\r?\n", helper)))
-  return t
+    io.popen("shared-state insert nodes_and_links", "w"):write(json.stringify(_nodes_and_links()))
```

Comment:
This is a side effect, so it shouldn't be bounded to the success of the function.
If the popen fails, it will make the function fail, right?

Reply:
If popen fails (like calling `io.popen("false")`) execution continues as nothing happened. So It is not an issue.